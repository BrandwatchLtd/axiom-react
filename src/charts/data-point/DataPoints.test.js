import React from 'react';
import renderer from 'react-test-renderer';
import { DataPoints, DataPoint } from 'bw-axiom';

const getComponent = (props = {}) =>
  renderer.create(
    <DataPoints size="1rem" { ...props }>
      <DataPoint color="blue" />
      <DataPoint color="pink" />
      <DataPoint color="amber" />
    </DataPoints>
  );

describe('DataPoints', () => {
  it('renders with defaultProps', () => {
    const component = getComponent();
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with onClick', () => {
    const component = getComponent({ onClick: () => {} });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
