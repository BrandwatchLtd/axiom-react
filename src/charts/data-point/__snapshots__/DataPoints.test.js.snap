// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataPoints it renders with type grouped 1`] = `
<svg
  className="ax-data-points"
  style={
    Object {
      "height": "1rem",
    }
  }
  viewBox="0 0 40 24"
>
  <circle
    className="ax-data-point ax-data-point--blue"
    cx={12}
    cy={12}
    r={12.75}
    strokeWidth={1.5}
  />
  <circle
    className="ax-data-point ax-data-point--pink"
    cx={20}
    cy={12}
    r={12.75}
    strokeWidth={1.5}
  />
  <circle
    className="ax-data-point ax-data-point--amber"
    cx={28}
    cy={12}
    r={12.75}
    strokeWidth={1.5}
  />
</svg>
`;

exports[`DataPoints it renders with type stacked 1`] = `
<svg
  className="ax-data-points"
  style={
    Object {
      "height": "1rem",
    }
  }
  viewBox="0 0 24 24"
>
  <circle
    className="ax-data-point ax-data-point--blue"
    cx={12}
    cy={12}
    r={12}
    strokeWidth={0}
  />
  <circle
    className="ax-data-point ax-data-point--pink"
    cx={12}
    cy={12}
    r={8}
    strokeWidth={0}
  />
  <circle
    className="ax-data-point ax-data-point--amber"
    cx={12}
    cy={12}
    r={4}
    strokeWidth={0}
  />
</svg>
`;

exports[`DataPoints renders with defaultProps 1`] = `
<svg
  className="ax-data-points"
  style={
    Object {
      "height": "1rem",
    }
  }
  viewBox="0 0 24 24"
>
  <circle
    className="ax-data-point ax-data-point--blue"
    cx={12}
    cy={12}
    r={12}
    strokeWidth={0}
  />
  <circle
    className="ax-data-point ax-data-point--pink"
    cx={12}
    cy={12}
    r={8}
    strokeWidth={0}
  />
  <circle
    className="ax-data-point ax-data-point--amber"
    cx={12}
    cy={12}
    r={4}
    strokeWidth={0}
  />
</svg>
`;
